src/main/kotlin/open/tresorier/services/OperationService.kt

```diff
@@ -57,4 +60,58 @@ class OperationService(private val operationDao: IOperationDao, private val auth
         authorizationService.cancelIfUserIsUnauthorized(person, budget)
         return operationDao.findByBudget(budget)
     }
-}
+    fun openAndReadOfxFile(person: Person, account: Account, fileOfx: String) {
+        val formattedFile: String = fileOfx.replace("\n", "")
+        val doubleFormattedFile: String = formattedFile.replace("\r", "")
+        this.separateOperationsOfOfxFile(person, account, doubleFormattedFile)
+    }
+    fun separateOperationsOfOfxFile(person: Person, account: Account, ofxFile: String) : List<String> {
+        authorizationService.cancelIfUserIsUnauthorized(person, account)
+        var operationList = mutableListOf<String>()
+        var startTag = ofxFile.indexOf("<STMTTRN>")
+        var closeTag = ofxFile.indexOf("</STMTTRN>") + 10
+        var lastTag = ofxFile.lastIndexOf("</STMTTRN>") + 10
+        while (closeTag !== lastTag) {
+            operationList.add(ofxFile.substring(startTag, closeTag))
+            startTag = closeTag
+            closeTag = ofxFile.indexOf("</STMTTRN>", startTag) + 10
+        }
+        if (startTag !== -1) {
+            operationList.add(ofxFile.substring(startTag, closeTag))
+        }
+        this.passInEachOperation(person, account, operationList)
+        return operationList
+    }
+    fun passInEachOperation(person: Person, account: Account, operationList: List<String>) {
+        operationList.forEach {
+            this.createOperationFromOFX(person, account, it)
+        }
+    }
+    fun createOperationFromOFX(person: Person, account: Account, operation: String): Operation {
+        // <TRNTYPE> balise de la nature de l'opération
+        var startElement = operation.indexOf("<TRNTYPE>") + 9
+        var endElement = operation.indexOf("<", startElement)
+        val type = operation.substring(startElement, endElement)
+        // <DTPOSTED> balise de la date
+        startElement = operation.indexOf("<DTPOSTED>") + 10
+        endElement = operation.indexOf("<", startElement)
+        val date = operation.substring(startElement, endElement)
+        // <TRNAMT> balise du montant
+        startElement = operation.indexOf("<TRNAMT>") + 9
+        endElement = operation.indexOf("<", startElement)
+        val euro = operation.substring(startElement, endElement)
+        // <MEMO> balise du mémo
+        startElement = operation.indexOf("<MEMO>") + 6
+        endElement = operation.indexOf("<", startElement)
+        val memo = operation.substring(startElement, endElement)
+        // formatage des données récupérées
+        val day = Day.createFromComparable(Integer.parseInt(date))
+        val cent: String = euro.replace(",", "")
+        var amount: Int = Integer.parseInt(cent)
+        if (type == "DEBIT") {
+            amount *= -1
+        }
+        val operationCreated = this.create(person, account, day, null, amount, memo)
```

Comment:
this both parse and add to the db, I think you could have a parse and save function that, for each ofxOperation would call the parser, get an operation in return then add it to the db